recipes/osgearth/all/conanfile.py

```diff
@@ -0,0 +1,314 @@
+            self.options.with_draco = False
+
+    def requirements(self):
+
+        self.requires("opengl/system")
+        self.requires("gdal/3.3.1")
+        self.requires("openscenegraph/3.6.5")
+        self.requires("libcurl/7.79.1")
+
+        self.requires("libtiff/4.3.0", override=True)
+        self.requires("openssl/1.1.1l", override=True)
+
+        # if self.options.build_triton_nodekit:
+        #     self.requires("triton_nodekit")
+        # if self.options.build_silverlining_nodekit:
+        #     self.requires("silverlining_nodekit")
+        if self.options.build_leveldb_cache:
+            self.requires("leveldb/1.22")
+        if self.options.build_rocksdb_cache:
+            self.requires("rocksdb/6.20.3")
+        if self.options.build_zip_plugin:
+            self.requires("libzip/1.7.3")
+            self.requires("zstd/1.4.9")  # override
+        if self.options.with_geos:
+            self.requires("geos/3.9.1")
+        if self.options.with_sqlite3:
+            self.requires("sqlite3/3.36.0")
+        if self.options.with_draco:
+            self.requires("draco/[>1.3.6]")
+        # if self.options.with_basisu:
+        #     self.requires("basisu")
+        # if self.options.with_glew:
+        #     self.requires("glew/2.2.0")
+        if self.options.with_protobuf:
+            self.requires("protobuf/3.17.1")
+        if self.options.with_webp:
+            self.requires("libwebp/1.2.0")
+
+    def _patch_sources(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
+
+        for package in ("Draco", "GEOS", "LevelDB", "OSG", "RocksDB", "Sqlite3", "WEBP"):
+            # Prefer conan's find package scripts over osgEarth's
+            os.unlink(os.path.join(self._source_subfolder, "CMakeModules", "Find{}.cmake".format(package)))
+
+        if os_info.is_windows:
+            tools.replace_in_file(os.path.join(self._source_subfolder, "CMakeLists.txt"),
+                                  "PROJECT(OSGEARTH)",
+                                  "PROJECT(OSGEARTH)\n add_compile_definitions(NOGDI)")
+
+        if os_info.detect_windows_subsystem() == "MSYS2":
+            tools.replace_in_file(os.path.join(self._source_subfolder,
+                                               "osgEarth", "src", "osgEarthDrivers", "fastdxt", "dxt.cpp"),
+                                  "tmpBuf = (byte*)memalign(16, width*height*4);",
+                                  "tmpBuf = (byte*)__mingw_aligned_malloc(width*height*4,16);")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version],
+                  destination=self._source_subfolder, strip_root=True)
+
+        self._patch_sources()
+
+    def _configured_cmake(self):
+        if hasattr(self, "_cmake"):
+            return self._cmake
+
+        generator_ = None
+        if os_info.detect_windows_subsystem() == "MSYS2":
```

Comment:
```suggestion
        if os_info.detect_windows_subsystem() == "MSYS2":
```
I don't think it's needed, you can configure `CONAN_CMAKE_GENERATOR` locally, if needed